### ISSUE TITLE
[🐛BUG] Unused objects in FEARec

### ISSUE THREAD

**Issue by jackmedda** [RUCAIBox/RecBole#1947]
I am not aware of how FEARec works, so I preferred to open an issue instead of a pull request.
I found some objects that were unused.

Line 436: I found a dropout layer not being used in the HybridAttention module with a hardcoded dropout probability, but I don't know if it should be actually used.
Line 616: q_ftt_box is created as follows:
```python
q_fft_box = torch.zeros(
    B, H, E, len(self.q_index), device=q_fft.device, dtype=torch.cfloat
)
```
but at the next line (617) it is immediately overwritten as:
```python
q_fft_box = q_fft[:, :, :, self.q_index]
```
it seems like 616 should be removed or line 617 should use q_ftt_box instead of q_ftt, but I'm not an expert of such model.

Similarly, it happens also at lines 621, 655, 665, 675

**Comment by TayTroye** [comment]
@jackmedda 
Hello！We keep the same implementation as the original [code](https://github.com/sudaada/FEARec).
